### PR TITLE
Added graceful handling of empty arrays.

### PR DESCRIPTION
**Pull request by thinkingmedia**
Fixes some unexpected errors when trying to output a table for an empty array, or an incorrectly structured array.

### CODE REVIEW

File: src/Shell/Helper/TableHelper.php [hunk 1]
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if(empty($rows) || !is_array($row)) {
```
Comment (chinpei215) [inline src/Shell/Helper/TableHelper.php]: Typo. `empty()` is not suitable to evaluate variables. Use `!$rows` instead. See #8397

File: src/Shell/Helper/TableHelper.php [hunk 2]
```diff
@@ -98,6 +102,10 @@ protected function _render($row, $widths, $options = [])
      */
     public function output($rows)
     {
+        if(empty($rows) || !is_array($rows)) {
```
Comment (chinpei215) [inline src/Shell/Helper/TableHelper.php]: Not typo, but use `!$rows` instead.

File: src/Shell/Helper/TableHelper.php [hunk 3]
```diff
@@ -106,6 +114,10 @@ public function output($rows)
             $this->_render(array_shift($rows), $widths, ['style' => $config['headerStyle']]);
             $this->_rowSeparator($widths);
         }
+    
+        if(empty($rows)) {
```
Comment (chinpei215) [inline src/Shell/Helper/TableHelper.php]: Same as above.

File: src/Shell/Helper/TableHelper.php (hunk 1 above)
Comment (markstory) [inline src/Shell/Helper/TableHelper.php]: Or count() if you know it is an array already.

**Comment by markstory** [comment]
Any way we could get tests for these scenarios ?

**Comment by thinkingmedia** [comment]
What if it uses type hinting instead?

I only used empty because I didn't want to change the function signature, but it would be better with `output(array $row)` as that addresses the above comments.

@markstory arg.... okay.

**Comment by ADmad** [comment]
Existing tests are also failing and there are CS errors.

**Comment by ADmad** [comment]
@thinkingmedia ping

**Comment by thinkingmedia** [comment]
I'll take a look at this today. Been sick.

File: src/Shell/Helper/TableHelper.php [hunk 4]
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if(!$row || !is_array($row)) {
```
Comment (lorenzo) [inline src/Shell/Helper/TableHelper.php]: `empty($rows)` or `count($rows)` would be shorter. Additionally, if rows is not an array it has the nice side effect that you will get an error :)

**Comment by thinkingmedia** [comment]
If everything passes then I'm finished.

File: src/Shell/Helper/TableHelper.php [hunk 5]
```diff
@@ -78,6 +78,10 @@ protected function _rowSeparator($widths)
      */
     protected function _render($row, $widths, $options = [])
     {
+        if (!is_array($row) || count($row) === 0) {
```
Comment (lorenzo) [inline src/Shell/Helper/TableHelper.php]: what's the need for checking if it is an array?

Comment (thinkingmedia) [inline src/Shell/Helper/TableHelper.php]: there was no parameter hint in the original code, and I didn't want to change the function signature.

Now that I just wrote the above comment. I see that's silly. I'll add type hinting to the method and remove `is_array`

**Pull request merged**